rust/kernel/lib.rs

```diff
@@ -28,12 +28,20 @@
 #![feature(ptr_metadata)]
 #![feature(receiver_trait)]
 #![feature(unsize)]
+// TODO resolve discussion
+#![feature(new_uninit)]
+#![feature(type_alias_impl_trait)]
+// these are already stabilized in newer versions
+#![feature(explicit_generic_args_with_impl_trait)]
 
 // Ensure conditional compilation based on the kernel configuration works;
 // otherwise we may silently break things like initcall handling.
 #[cfg(not(CONFIG_RUST))]
 compile_error!("Missing kernel configuration for conditional compilation");
 
+// allow macros to refer to `::kernel`.
+extern crate self as kernel;
```

Comment:
I don't see any references to it in any macro.

Reply:
my `pin_data` macro uses it, I should clarify that this only applies to proc-macros.